doc/meson.build

```diff
@@ -0,0 +1,81 @@
+# SPDX-License-Identifier: LGPL-2.1-or-later
+
+if want_mans == true
+    PAGES = [
+        ['lxc', '7'],
+        ['lxc.conf', '5'],
+        ['lxc.container.conf', '5'],
+        ['lxc.system.conf', '5'],
+        ['lxc-update-config', '1'],
+        ['lxc-usernet', '5'],
+        ['lxc-user-nic', '1'],
+        ['lxc-usernsexec', '1'],
+    ]
+
+    if want_tools == true
+        PAGES += [
+            ['lxc-attach', '1'],
+            ['lxc-autostart', '1'],
+            ['lxc-cgroup', '1'],
+            ['lxc-checkconfig', '1'],
+            ['lxc-checkpoint', '1'],
+            ['lxc-config', '1'],
+            ['lxc-console', '1'],
+            ['lxc-copy', '1'],
+            ['lxc-create', '1'],
+            ['lxc-destroy', '1'],
+            ['lxc-device', '1'],
+            ['lxc-execute', '1'],
+            ['lxc-freeze', '1'],
+            ['lxc-info', '1'],
+            ['lxc-ls', '1'],
+            ['lxc-monitor', '1'],
+            ['lxc-snapshot', '1'],
+            ['lxc-start', '1'],
+            ['lxc-stop', '1'],
+            ['lxc-top', '1'],
+            ['lxc-unfreeze', '1'],
+            ['lxc-unshare', '1'],
+            ['lxc-wait', '1'],
+        ]
+    endif
+
+    if want_pam_cgroup == true
+        PAGES += [
+            ['pam_cgfs', '8'],
+        ]
+    endif
+
+    # Common files.
+    configure_file(
+        configuration: conf,
+        input: 'common_options.sgml.in',
+        output: 'common_options.sgml')
+
+    configure_file(
+        configuration: conf,
+        input: 'see_also.sgml.in',
+        output: 'see_also.sgml')
+
+    # Config
+    docconf = configuration_data()
+    docconf.set('docdtd', '"-//OASIS//DTD DocBook XML" "http://www.oasis-open.org/docbook/xml/4.5/docbookx.dtd"')
+    docconf.set('builddir', '.')
+    docconf.set('LXC_GENERATE_DATE', generate_date)
+    docconf.set_quoted('PACKAGE_VERSION', meson.project_version())
+
+    # Initial templating.
+    foreach page : PAGES
+        sgml = configure_file(
+            configuration: docconf,
+            input: page[0] + '.sgml.in',
+            output: page[0] + '.sgml')
+
+        configure_file(
+            input: sgml,
+            output: page[0] + '.' + page[1],
+            command: ['docbook2x-man', '--encoding=UTF-8', page[0] + '.sgml'],
```

Comment:
https://github.com/lxc/lxc/blob/9647df5464acd2e71e7b40e185d6a6cb842dd89f/configure.ac#L228-L238

In order to have feature parity with the autotools build, you need to check all these alternative names too. e.g.

```meson
db2xman = find_program('docbook2X2man', 'docbook2x-man', 'db2x_docbook2man', 'docbook2man', 'docbook-to-man')

By the way, why use configure_file here instead of custom_target?